### ISSUE TITLE
Does not write values to data base

### ISSUE THREAD

**Issue by overpaul** [gjr80/wee_import#5]
for some reason on a csv conversion it will only write interval  usUnits 'dateTime to the database.
This is true of the test file and my csv file.
Looks like the line _rec[_field] = _temp is missing or there is something else blocking the writing of the data fields (my testing could not find a reason for this)
WU conversions seem to work
got it to work by adding  _rec[_field] = _temp

around:
                        # field by field unit conversions
                        if _units is None:
                            _temp_vt = ValueTuple(_temp,
                                                  self.map[_field]['units'],
                                                  weewx.units.obs_group_dict[_field])
                            _conv_vt = convertStd(_temp_vt, unit_sys)
#                            print ('conv VT %s' % _conv_vt.value)
                            _rec[_field] = _conv_vt.value
                        _rec[_field] = _temp

**Comment by gjr80** [comment]
Thanks, not sure how that one got through quality control. Fixed at commit https://github.com/gjr80/wee_import/commit/a8f598645243e08ba728ff231243d8f6166e0d2b. Should work fine now.

**Comment by overpaul** [comment]
Thanks for fixing it.
I am learning Python so I learned a lot tracking down the bug. Having your
test case and seeing that it failed, was a great help as I first thought my
setup was wrong.
Your app really solved a problem for me as I was moving from VWS on the PC
to Weewx on the Pi and wanted to move my historical data over.


*Paul Oversmith*

E-Mail: pgo@overweb.com
Phone: (941) 266-3600

**Comment by gjr80** [comment]
Sorry for the delay in replying. Often getting your hands in and tracking down and dealing with a problem is the best way to learn. Must admit, there was some hesitation about producing `wee_import` at first as the original user request was to import data from WU (which is fraught with danger, mainly due to WU idiosyncracies), but your use of `wee_import` to import from VWS is a good use case. Thanks.

**Comment by overpaul** [comment]
Actually, I did both - I used VWS as the primary input and WU for the times
where I had a gap in coverage because the computer running VWS was off. I
had WeatherLink updating the data to WU also, so I did have WU data points
when VWS was not on.

One suggestion I would make is to add an option to list the converted
records when doing a dry-run. I put a print statement in after the data was
converted  to show what would be transferred and this helped me to check
the dates and field selection options.

Your effort to write wee_import was most appreciated as it solved a real
conversion problem I had.  Thank you.


*Paul Oversmith*

E-Mail: pgo@overweb.com
Phone: (941) 266-3600

**Comment by gjr80** [comment]
Maybe your approach with WU did not suffer from the issues I have experienced in testing. I have found that despite their being a full days worth of data on WU, importing a whole day seldom returns all 288 records (at 5 minute archive intervals). Usually returns 280-something. Plays havoc with the code if you have it set to derive the interval from the data.

Good idea re `--dry-run`, I have a `--verbose` option that I was looking at the other day trying to work out what o get it to display, this is probably a good item to put under its control.

**Comment by overpaul** [comment]
I did have problems with WU intervals.  This was compounded by having both
WU and VWS update WU.
Since there wasn't too much data I edited the SQF file to remove duplicate
updates and adjusted the interval to 5 minutes when it was 4 or 6 minutes
and removed the 2 minute ones when there was it was next to a 3 minute one
(which I made 5 minutes)
This probably isn't all that accurate, but at least I have consistent
intervals in weewx.

It also looked like it did not get rainfall  from WU.  I discovered this
after I merged the WU and VWS data and decided not to go back and try to
figure out what went wrong in data that is a year old.

again thanks for building wee_imprort.  It really was a great help for me.


*Paul Oversmith*

E-Mail: pgo@overweb.com
Phone: (941) 266-3600
